src/test/java/org/broadinstitute/consent/http/service/dao/DacServiceDAOTest.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.RandomUtils;
+import org.broadinstitute.consent.http.db.DAOTestHelper;
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.models.Dataset;
+import org.broadinstitute.consent.http.models.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DacServiceDAOTest extends DAOTestHelper {
+
+  private DacServiceDAO serviceDAO;
+
+  @BeforeEach
+  void setUp() {
+    serviceDAO = new DacServiceDAO(jdbi);
+  }
+
+  @Test
+  void testDeleteDac() {
+    User superUser = createUser();
+    // Create DACs and all associated objects subject to deletion
+    List<Dac> dacs = createMockDACs();
+    dacs.forEach(dac -> {
+      // DAC
+      int dacId = dacDAO.createDac(RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10), RandomStringUtils.randomAlphabetic(10),  new Date());
+      // DAA
+      int daaId = daaDAO.createDaa(superUser.getUserId(), new Date().toInstant(), superUser.getUserId(), new Date().toInstant(), dacId);
+      // DAC-DAA Association
+      daaDAO.createDacDaaRelation(dacId, daaId);
+      // Library Card User with Institution
+      User lcUser = createUser();
+      int institutionId = institutionDAO.insertInstitution(
+          RandomStringUtils.randomAlphabetic(10),
```

Comment:
The `RandomStringUtils.randomAlphabetic(10)` aren't very meaningful, it may be worthwhile replacing these with `itDirectorName`, `itDirectorEmail`, `institutionUrl`, and so on so we can see what these values represent.